hal/src/gpio/mod.rs

```diff
@@ -16,7 +16,8 @@
 //!
 //! ## Features
 //!
-//! - Converting between pin modes no longer requires access to the `Port` type.
+//! - Converting between pin modes no longer requires access to the
```

Comment:
Let's just get rid of this whole reference to the old GPIO module - it was useful when we had two GPIO versions in the HAL but now only of historical interest